### PR TITLE
removing annotation default editor UI

### PR DESCRIPTION
**Pull request by btburton42**
• importing ui.main and customizing it
  • rewiring our module to use a class constructor
  • passing in as window object until we get webpack working
  • also cleaned up some of the SCSS to make it more BEM compliant

### CODE REVIEW

**Comment by mngyuan** [comment]
Sorry for the wait! Some thoughts:
- :+1: the ES6 class syntax
- Does use of the gerund mean the work is in progress or finished and included in this PR?
- If I understand correctly, this PR, along with some tidying work, removes the comment bubble that appears when you click to make an annotation. It looks like an unintentional side effect is that I can click to annotation anything now, including when I click on buttons and radio buttons, which I think is undesirable (unless this is intended).

**Comment by btburton42** [comment]
hi kevin, no worries about the wait. 

so, to preface, this work basically copies whole-sale the editor.js from annotator core. my next task will be to add webpack into the build and move the whole syntax into es6 and start leveraging NPM and some imports from annotator core. in the meantime, i'm posting this WIP not because it is a finished product but because it's necessary to unblock UI work and it gets the learnings from my call with the annotator creator out of my head and into some code. 

in short, there is still a lot to do. viewer will have to be ported, and maybe the adder, but this paves the way forward.

if you're giving a :+1: i'll go ahead and merge. otherwise keep up the feedback and questions. thanks.

File: src/ui-main.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    var offset = {top: 0, left: 0};
+
+    if ($(body).css('position') !== "static") {
+        offset = $(body).offset();
+    }
+
+    return {
+        top: event.pageY - offset.top,
+        left: event.pageX - offset.left
+    };
```
Comment (mngyuan) [inline src/ui-main.js:24]: Checking if body's position is static seems hacky here... when would body's position change?

File: src/ui-main.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+// annotationFactory returns a function that can be used to construct an
+// annotation from a list of selected ranges.
+function annotationFactory(contextEl, ignoreSelector) {
+    return function (ranges) {
+        var text = [],
+            serializedRanges = [];
+
+        for (var i = 0, len = ranges.length; i < len; i++) {
+            var r = ranges[i];
+            text.push(String.prototype.trim(r.text()));
```
Comment (mngyuan) [inline src/ui-main.js:36]: `text.push(r.text().trim())` ?

File: src/ui-main.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+ *
+ *      An array of viewer extensions. See the
+ *      :class:`~annotator.ui.viewer.Viewer` documentation for details of viewer
+ *      extensions.
+ *
+ */
+function main(options) {
+    if (typeof options === 'undefined' || options === null) {
+        options = {};
+    }
```
Comment (mngyuan) [inline src/ui-main.js:81]: `if (!options) { options = {}; }`
should be the same, or my favorite
`options || (options = {});`

File: src/ui-main.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+                    s.interactionPoint = mousePosition(event);
+                    s.adder.load(annotation, s.interactionPoint);
+                } else {
+                    s.adder.hide();
+                }
+            }
+        });
+
+        s.viewer = new viewer.Viewer({
+            onEdit: function (ann) {
```
Comment (mngyuan) [inline src/ui-main.js:125]: ES6 should support the cleaner (IMO)

```
s.viewer = new viewer.Viewer({
  onEdit(ann) {
    ...
  },
  onDelete(ann) {
    ...
  },
  ...
});
```

File: src/ui-main.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+            s.highlighter.destroy();
+            s.textselector.destroy();
+            s.viewer.destroy();
+            removeDynamicStyle();
+        },
+
+        annotationsLoaded: function (anns) { s.highlighter.drawAll(anns); },
+        annotationCreated: function (ann) { s.highlighter.draw(ann); },
+        annotationDeleted: function (ann) { s.highlighter.undraw(ann); },
+        annotationUpdated: function (ann) { s.highlighter.redraw(ann); },
```
Comment (mngyuan) [inline src/ui-main.js:163]: for these functions too, same as my above comment

**Comment by mngyuan** [comment]
Feel free to merge, then, if this is blocking you. Made some comments on some small nit and style things that aren't critical but improve code quality IMO. Code quality is obviously subjective so open to suggestions.

:ship: it :+1:

**Comment by btburton42** [comment]
ha, just to be clear, none of the stuff in `ui-main.js` is my code, it's taken wholesale from annotator per the suggestions of the annotator creator. this stuff is :hankey: and it's absolutely a significant chunk of techdebt, but i'm comfortable with taking that on to unblock parallel work streams, and because its only 181 lines. eventually, either you or myself will rewrite this from scratch.

going to merge and we can discuss next steps on sunday.

**Comment by btburton42** [comment]
i'm also filing an issue to keep us honest and bound to removing this.
